doozer

```diff
@@ -307,12 +307,16 @@ def images_update_dockerfile(runtime, stream, version, release, repo_type, messa
 
     try:
         if push:
-            runtime.push_distgits()
+            res = runtime.push_distgits()
 
-        state.record_image_finish(lstate)
+            for name, r in res:
+                if r is not True:
+                    state.record_image_fail(lstate, name, r)
     except Exception as ex:
         lstate['status'] = state.STATE_FAIL
-        raise DoozerFatalError(ex.message)
+        raise DoozerFatalError(repr(ex))
```

Comment:
This seems fine to me, but there's a way to use both if you think it's better:
https://stackoverflow.com/a/45532289
```suggestion
        raise DoozerFatalError(getattr(ex, 'message', repr(ex)))
```